sdk/dynamicconfig/sdk/metric/controller/notifier/servicereader.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
```

Comment:
I'm sorry this is a comment for something already comitted, but when calling readConfig, I notice that you [open a new connection with each call](https://github.com/vmingchen/opentelemetry-go-contrib/blob/c740d0aed0244a882808828f10f065b4e357ccb5/sdk/dynamicconfig/sdk/metric/controller/notifier/servicereader.go#L69). To save on overhead, what do you think about opening a connection as the pusher is instantiated, and keeping a live connection through the lifetime of the pusher? I believe that's the strategy adopted by components in the collector (and potentially in the sdk?) as well.

Reply:
Hmm. This is a good point.

I think keeping a live connection for the collector may make more sense, in that in a real production system, presumably fairly large amounts of data are frequently getting exported and then sent to the metrics back-end.

In contrast, realistically, it's probably not urgent for us to check for and apply new configs, and most processes will probably only check every 10 minutes or so. We don't really care much (within reason) about the latency of `readConfig`, so it might not be worth the overhead to the configuration service to keep a live connection.

Reply:
This is an interesting question. It may depend on whether you use multiple resources per process (I believe resource is tied to the accumulator?). If you have a large number of resources, you might need to have a shared connection for them, because you will have frequent readConfig calls and wouldn't want to open up a new connection for each one. I'm not sure if exporting metrics for a large number of resources is a planned use case though.